src/zjs_fs.c

```diff
@@ -409,19 +413,27 @@ static jerry_value_t zjs_read(const jerry_value_t function_obj,
         return zjs_error("offset + length overflows buffer");
     }
 
-    // if a position was specified, seek to it before writing
-    if (fs_seek(&handle->fp, (uint32_t)position, SEEK_SET) != 0) {
+    if (handle->mode == MODE_A_PLUS) {
+        // mode is a+, seek to read position
+        if (fs_seek(&handle->fp, handle->rpos, SEEK_SET) != 0) {
+            return zjs_error("error seeking to position");
+        }
+    }
+    // if a position was specified, seek to it before reading
+    if (fs_seek(&handle->fp, (uint32_t)position, SEEK_CUR) != 0) {
         return zjs_error("error seeking to position");
     }
+    handle->rpos += position;
```

Comment:
Make sure to do this only in the case where you do the seek above, and again it's rpos = position I think, not += (absolute, not relative).